src/app/profile/overview/spaces/spaces.component.spec.ts

```diff
@@ -65,6 +60,12 @@ describe('SpacesComponent', () => {
   });
 
   describe('#initSpaces', () => {
+
+    // ensure the component is not left in a loading state
+    afterEach(() => {
+      expect(component.loading).toEqual(false);
+    });
```

Comment:
Feels weird to put a test assertion in an afterEach like that, but I guess the intent makes sense. What happens if this fails though, since the `expect` isn't within an `it`? Does it accurately report which `it` block caused the failure?

Reply:
Yeah, it displays the test that it failed on, and what value it was expecting. This afterEach acts within the `describe('#initSpaces', ... )` block it is in, so it won't apply to every test in the spec file. For example, I slipped a `component.loading = true` into the last test of that block to force a failure, and the output was:

```
FAILED TESTS:
  SpacesComponent
    #initSpaces
      ✖ should log an error if the context or context.user is empty
        PhantomJS 2.1.1 (Linux 0.0.0)
      Expected true to equal false.
```

Reply:
Cool, good to know. If you add an `it` inside the `afterEach`/around the `expect`, maybe the error reporting would be even clearer?

Reply:
Neat, if you put code inside of an `it` inside of an `afterEach` it doesn't seem to run. The `expect` wasn't hit, along with a bunch of logs that I threw in there.

Reply:
Interesting. It's fine as-is anyway, I was just curious as I don't tend to use `afterEach`.